impl/src/main/kotlin/nl/nuts/consent/bridge/corda/CordaManagedConnection.kt

```diff
@@ -40,60 +41,113 @@ import java.net.Socket
 class CordaManagedConnection(val consentBridgeRPCProperties: ConsentBridgeRPCProperties) : EventedConnection<CordaRPCConnection>() {
 
     private val logger: Logger = LoggerFactory.getLogger(this::class.java)
-    private var connection: CordaRPCConnection? = null
-    private var tryToConnect:Boolean = false
-    private var terminate:Boolean = false
-    private var retryCount = consentBridgeRPCProperties.retryCount
-    private var noConnectionReason: String? = "starting up"
-    private var watchdog: Thread? = null
+    private var cordaConnectionWatchdog: CordaConnectionWatchdog? = null
+    //private var retryCount = consentBridgeRPCProperties.retryCount
 
-    var name: String = "unknown corda"
+    var name: String? = "unknown corda"
 
-    private fun startThread() {
-        watchdog = Thread {
-            while(!terminate) {
-                synchronized(this) {
-                    if (connection != null) {
-                        try {
-                            val nodeInfo = connection?.proxy?.nodeInfo()
-                            require(nodeInfo?.legalIdentitiesAndCerts?.isNotEmpty() ?: false)
-                        } catch (e: IllegalArgumentException) {
-                            noConnectionReason = e.message
-                            logger.error("Corda RPC connection lost for $name: ${e.message}")
-                            connection = null
-                            this.onDisconnected()
-                        }
-                    } else if (tryToConnect) {
-                        try {
-                            connectToCorda()
-                            logger.info("Corda RPC connection established for $name")
-                            this.onConnected()
-                        } catch (e: IOException) {
-                            noConnectionReason = e.message // if host is unreachable
-                        } catch (secEx: ActiveMQSecurityException) {
-                            // Incorrect credentials
-                            noConnectionReason = secEx.message
-                        } catch (ex: RPCException) {
-                            noConnectionReason = ex.message
+    class CordaConnectionWatchdog(var managedConnection: CordaManagedConnection, var delay: Long, var onConnected: () -> Unit, var onDisconnected: () -> Unit, var onError: () -> Unit = {}): Runnable {
```

Comment:
I think this class needs some documentation since it's quite a complex piece. What it solves, how it does so and how it should be used. Maybe some explanation on the implementation of run(...)

(would help in at least my understanding)